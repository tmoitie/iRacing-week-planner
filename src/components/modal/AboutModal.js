// @flow

import * as React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import BuyACoffee from '../BuyACoffee';
import Modal from './Modal';
import changelog from '../../data/changelog';
import contributors from '../../data/contributors.json';

import styles from '../../styles/main.module.scss';

type Props = {
  onClose: () => void,
  isOpen: boolean,
};

export default function AboutModal({ onClose, isOpen }: Props) {
  const { t } = useTranslation();

  return (
    <Modal onClose={onClose} isOpen={isOpen} title={t('About')} doneAction={onClose} id="about-modal">
      <div className={styles['container-fluid']}>
        <p>
          <Trans>
            This tool was created by
            {' '}
            <a href="https://twitter.com/tmoitie" target="_blank" rel="noreferrer">
              tmoitie
            </a>
            {' '}
            (
            <a
              href="http://members.iracing.com/membersite/member/CareerStats.do?custid=69636"
              target="_blank"
              rel="noreferrer"
            >
              Tom Moitié
            </a>
            {' '}
            on iRacing).
            Feel free to contact me via twitter or iRacing if you have any feedback or questions. The code is hosted
            publicly on
            {' '}
            <a href="https://github.com/tmoitie/iRacing-week-planner" target="_blank" rel="noreferrer">Github</a>
            . Thanks!
          </Trans>
        </p>

        <p>
          <Trans>
            This is now costing me a few quid a month to run, so if you like using this tool, please feel free
            to
            {' '}
            <a
              href="https://www.buymeacoffee.com/tmoitie"
              target="_blank"
              rel="noreferrer"
            >
              buy me a coffee
            </a>
            .
          </Trans>
        </p>

        <p>
          <BuyACoffee />
        </p>

        <h3>{t('Contributors')}</h3>
        <ul className={styles.row}>
          {contributors.map((contributor) => (
            <li className={`${styles['col-md-3']} ${styles['col-sm-4']} ${styles['col-xs-6']}`} key={contributor.id}>
              <a href={contributor.html_url} target="_blank" rel="noreferrer">
                {contributor.login}
              </a>
            </li>
          ))}
          <li className={`${styles['col-md-3']} ${styles['col-sm-4']} ${styles['col-xs-6']}`}>
            Yigit Aydin
          </li>
          <li className={`${styles['col-md-3']} ${styles['col-sm-4']} ${styles['col-xs-6']}`}>
            Rafał Noskowicz
          </li>
          <li className={`${styles['col-md-3']} ${styles['col-sm-4']} ${styles['col-xs-6']}`}>
            Rafał Znany
          </li>
        </ul>

        <h3>{t('Changelog')}</h3>
        {changelog.map((dayItem) => (
          <div key={dayItem.date}>
            <h4>
              {t(
                '{{date, datetime}}',
                {
                  date: dayItem.date.local().toDate(),
                  formatParams: {
                    date: { dateStyle: 'long' },
                  },
                },
              )}
            </h4>
            <ul>
              {dayItem.items.map((changeItem) => <li key={changeItem}>{changeItem}</li>)}
            </ul>
          </div>
        ))}
      </div>
    </Modal>
  );
}
