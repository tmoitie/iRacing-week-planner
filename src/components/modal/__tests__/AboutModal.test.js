import { describe, test } from '@jest/globals';
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import AboutModal from '../AboutModal';

describe('components/modal/AboutModal', () => {
  test('renders closed', async () => {
    const onClose = jest.fn(() => {});
    render(
      <AboutModal isOpen={false} onClose={onClose} />,
    );
    expect(screen.queryByTestId('about-modal')).not.toBeTruthy();
  });

  test('renders open', async () => {
    const user = userEvent.setup();
    const onClose = jest.fn(() => {});
    render(
      <AboutModal isOpen onClose={onClose} />,
    );

    expect(await screen.findByTestId('about-modal')).toMatchSnapshot();

    await user.click(screen.getByText('Close'));
    expect(onClose).toHaveBeenCalled();
  });
});
