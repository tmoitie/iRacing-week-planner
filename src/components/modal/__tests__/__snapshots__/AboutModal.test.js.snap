// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/modal/AboutModal renders open 1`] = `
<div
  class="Modal__Bootstrap modal-dialog modal-lg"
  data-testid="about-modal"
>
  <div
    class="modal-content"
  >
    <div
      class="modal-header"
    >
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <span
          class="glyphicon glyphicon-remove"
        />
      </button>
      <h4
        class="modal-title"
      >
        About
      </h4>
    </div>
    <div
      class="modal-body"
      style="max-height: 55vh; overflow-y: auto;"
    >
      <div
        class="container-fluid"
      >
        <p>
          This tool was created by
           
          <a
            href="https://twitter.com/tmoitie"
            rel="noreferrer"
            target="_blank"
          >
            tmoitie
          </a>
           
          (
          <a
            href="http://members.iracing.com/membersite/member/CareerStats.do?custid=69636"
            rel="noreferrer"
            target="_blank"
          >
            Tom Moitié
          </a>
           
          on iRacing). Feel free to contact me via twitter or iRacing if you have any feedback or questions. The code is hosted publicly on
           
          <a
            href="https://github.com/tmoitie/iRacing-week-planner"
            rel="noreferrer"
            target="_blank"
          >
            Github
          </a>
          . Thanks!
        </p>
        <p>
          This is now costing me a few quid a month to run, so if you like using this tool, please feel free to
           
          <a
            href="https://www.buymeacoffee.com/tmoitie"
            rel="noreferrer"
            target="_blank"
          >
            buy me a coffee
          </a>
          .
        </p>
        <p>
          <button
            class="bmcButton"
            type="button"
          >
            <img
              alt="Buy me a coffee"
              src="https://cdn.buymeacoffee.com/buttons/bmc-new-btn-logo.svg"
            />
            <span
              style="margin-left: 5px;"
            >
              Buy me a coffee
            </span>
          </button>
        </p>
        <h3>
          Contributors
        </h3>
        <ul
          class="row"
        >
          <li
            class="col-md-3 col-sm-4 col-xs-6"
          >
            <a
              href="https://github.com/tmoitie"
              rel="noreferrer"
              target="_blank"
            >
              tmoitie
            </a>
          </li>
          <li
            class="col-md-3 col-sm-4 col-xs-6"
          >
            <a
              href="https://github.com/flomarin88"
              rel="noreferrer"
              target="_blank"
            >
              flomarin88
            </a>
          </li>
          <li
            class="col-md-3 col-sm-4 col-xs-6"
          >
            Yigit Aydin
          </li>
          <li
            class="col-md-3 col-sm-4 col-xs-6"
          >
            Rafał Noskowicz
          </li>
          <li
            class="col-md-3 col-sm-4 col-xs-6"
          >
            Rafał Znany
          </li>
        </ul>
        <h3>
          Changelog
        </h3>
        <div>
          <h4>
            December 16, 2015
          </h4>
          <ul>
            <li>
              Add ability to 'favorite' content with the star icon.
            </li>
            <li>
              Fix bug with disabled default content.
            </li>
          </ul>
        </div>
        <div>
          <h4>
            December 15, 2015
          </h4>
          <ul>
            <li>
              Add "Select All Oval/Road" to content choosers.
            </li>
            <li>
              Made free content un-unselectable.
            </li>
            <li>
              Made series clickable with popup showing all race weeks.
            </li>
            <li>
              Add licence class column to table.
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="modal-footer"
    >
      <button
        class="btn btn-primary"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
